specs/file_hierarchy_for_the_verification_of_os_artifacts.md

```diff
@@ -0,0 +1,639 @@
+
+The use of hardlinks in the VOA structure is discouraged to prevent confusion.
+However, VOA implementations are not required to check for the use of hardlinks.
+
+### Access library API considerations
+
+As [os] strings are relatively complex, a VOA access library may offer convenient APIs for enumeration/searching of _os_ directories (e.g. looking up a particular _os_ in the set of available _os_ strings, by searching by the _ID_ part of the [os-release] information).
+
+### Constraining verifiers
+
+A VOA library may allow applications to set constraints on successful verification.
+
+These constraints may include scenarios such as:
+
+- an application may require that _trust anchors_ must be present
+- an application may mandate the set of verifiers that must be present (either as _trust anchors_, or as _artifact verifiers_)
+
+### Verifier optimization
+
+When importing verifiers, a VOA library may support normalization operations on the verifier representation.
+For example, OpenPGP decryption and authentication component keys may be dropped, as they are not needed in any VOA context.
+
+### Masking verifiers
+
+The action of [masking] _signature verifiers_ should be able to distinguish between persistent and runtime directories (e.g. `/etc/voa/` and `/run/voa/`).
+By default, persistent locations should be preferred over runtime ones.
+
+### Threshold signing
+
+In some scenarios users of VOA may want to rely on verification schemes that require more than one valid digital signature for a given artifact.
+These scenarios are explicitly not part of VOA, but may be implemented using dedicated configuration-based approaches per [technology].
+
+### Retrieval of signature verifiers
+
+By default the _signature verifiers_ found in VOA are usually either provided by vendor updates to the operating system (e.g. in `/usr/share/voa/`) or are created by system administrators (e.g. in `/etc/voa/`).
+
+In certain scenarios, applications may want to retrieve additional or updated _signature verifiers_ from locations outside of the VOA hierarchy.
+These _signature verifiers_ should be placed in ephemeral runtime directories (e.g. `/run/voa/`).
+
+### Validity of signatures
+
+Each [technology] may handle validity differently.
+Generally, signature validity should be considered at the time of signature creation.
+The _signature verifier_ must be valid at creation time.
+If the _signature verifier_ has expired after signature creation, this does not impact signature validity.
+Only in the case of revocations, that indicate a key material compromise, all signatures by the key in question should be considered invalid, regardless of creation time.
+
+#### Use of Time Stamp Authority
+
+Some technologies support the use of _Time Stamp Authorities_ (e.g. based on the [time stamp protocol]).
+It is recommended to use timestamping services to validate signature creation time, where possible.
```

Comment:
I figure you want to emphasize *signature* here, because that's the key info here, no? i.e. not certificate timestamps, but signature timestamps.